Include build package in package.json for npm installs.

kaidjohnson opened the pull request:
While `xregexp-all.js` exists in the repo, it isn't included in the npm install. Adding the file to the package.json ensures npm installs provide a production-ready version for non-node applications.

This PR fixes #139

slevithan commented:
Thanks!

Can you clarify about removing `src` from `files`? Why is that necessary, and will the `src` directory still be included so that people can `require` individual modules?

kaidjohnson commented:
Yeah, I missed that rather crucial bit about supporting the modules therein. Sorry about that!

I've fixed this issue by updating the `main` to point to the built file now to ensure `require('xregexp');` pulls in the complete package. The `src` files are still included for incremental builds. Added the LICENSE for npm installs as well, per best practices.
